File: 3513.number-of-unique-xor-triplets-i.py
#
# @lc app=leetcode id=3513 lang=python3
#
# [3513] Number of Unique XOR Triplets I
#
# @lc code=start
class Solution:
    def uniqueXorTriplets(self, nums: List[int]) -> int:
        n = len(nums)
        xor_values = set()
        
        for i in range(n):
            for j in range(i, n):
                for k in range(j, n):
                    xor_val = nums[i] ^ nums[j] ^ nums[k]
                    xor_values.add(xor_val)
        
        return len(xor_values)
# @lc code=end

File: 3513.number-of-unique-xor-triplets-i.cpp
#
# @lc app=leetcode id=3513 lang=cpp
#
# [3513] Number of Unique XOR Triplets I
#

# @lc code=start
class Solution {
public:
    int uniqueXorTriplets(vector<int>& nums) {
        int n = nums.size();
        if (n == 1) return 1;
        if (n == 2) return 2;
        int l = 31 - __builtin_clz(n);
        return 1 << (l + 1);
    }
};
# @lc code=end
